src/middlewares/permissions.js

```diff
@@ -19,24 +31,34 @@ function createCheckPermission(environmentSecret, collectionName) {
     };
   }
 
-  function checkPermissionListAndSearch(request, response, next) {
-    const { searchToEdit } = request.query;
-    const renderingId = getRenderingIdFromUser(request.user);
-    const permissionName = searchToEdit ? 'searchToEdit' : 'list';
+  list() {
+    return (request, response, next) => {
+      const { searchToEdit } = request.query;
+      const permissionName = searchToEdit ? 'searchToEdit' : 'list';
 
-    return new PermissionsChecker(environmentSecret, renderingId, collectionName, permissionName)
-      .perform()
-      .then(next)
-      .catch((error) => {
-        logger.error(error.message);
-        next(httpError(403));
-      });
+      return this._checkPermission(permissionName)(request, response, next);
+    };
   }
 
-  return {
-    checkPermission,
-    checkPermissionListAndSearch,
-  };
+  export() {
+    return this._checkPermission('export');
+  }
+
+  details() {
+    return this._checkPermission('show');
```

Comment:
[Misc] Why did you change from `show` to `details` ?

Reply:
To be more explicit. Show means nothing to me, details is better (My point of view, feel free to contest it)